dbt_subprojects/daily_spellbook/models/evms/evms_contracts.sql

```diff
@@ -32,25 +32,29 @@
 ] %}
 
 SELECT *
-FROM (
+    FROM
+    (
         {% for contracts_model in contracts_models %}
         SELECT
-        '{{ contracts_model[0] }}' AS blockchain
-        , abi_id
-        , abi
-        , address
-        , "from"
-        , code
-        , name
-        , namespace
-        , dynamic
-        , base
-        , factory
-        , detection_source
-        , created_at
+            '{{ contracts_model[0] }}' AS blockchain
+            , abi_id
+            , abi
+            , address
+            , "from"
+            , code
+            , name
+            , namespace
+            , dynamic
+            , base
+            , factory
+            , detection_source
+            , created_at
+            , row_number() over (partition by address order by created_at desc) as duplicates_rank
```

Comment:
```suggestion
            , row_number() over (partition by blockchain, address order by created_at desc) as duplicates_rank
```

Reply:
if we're unique on these two fields, don't they both need added to partition in row number?

Reply:
You are right, I did the partition by address before union by chain.
But would be safer to partition by blockchain and address together.